circuit_utils/nodes.py

```diff
@@ -10,6 +10,12 @@ def __init__(self, value: str):
             self.value = 0
         elif value == 1 or value == '1':
             self.value = 1
+
+        # Final value for Ds wil be string "D" / "D'"
+        elif value == 'D' or value =='d':
+            self.value = 'D'
```

Comment:
should be using Value("D"), not "D"

Reply:
should I add this to the 1 and 0 cases as well?

Reply:
Yes. Do not set self.value to a character -- set it to a Value(1), Value('D'), etc